TypeScript: Use triple slash directives to avoid requiring users to include "dom" in their lib declarations

### Reproduction

Create a TS project which does not include "DOM" in the lib configuration, and a trivial SSR setup. 

```ts
import render from 'preact-render-to-string';
import { h } from 'preact';

const App = <div class="foo">content</div>;

console.log(render(App));
```

Repro: https://repl.it/repls/IllEmotionalCrypto#main.sh

### Expected Behavior

No compiler error

### Actual Behavior

TS yells about missing names:

```text
...

node_modules/preact/src/jsx.d.ts:926:22 - error TS2304: Cannot find name 'SVGUseElement'.

926     use: SVGAttributes<SVGUseElement>;
                           ~~~~~~~~~~~~~

Found 219 errors.
```

### Suggested solution

Add a triple slash directive which references the dom lib file in `jsx.d.ts`

```ts
// This reference allows TS users who are only using Preact for SSR to compile without including dom typings in their project. 
/// <reference lib="dom" />

type Defaultize<Props, Defaults> = ...
```

--- a/src/jsx.d.ts
+++ b/src/jsx.d.ts
@@ -1,3 +1,6 @@
+// Users who only use Preact for SSR might not specify "dom" in their lib in tsconfig.json
+/// <reference lib="dom" />
+
 type Defaultize<Props, Defaults> =
 	// Distribute over unions
 	Props extends any // Make any properties included in Default optional

